testhelpers/testhelpers.go

```diff
@@ -243,8 +274,24 @@ func ImageID(t *testing.T, repoName string) string {
 	return inspect.ID
 }
 
-func CreateSingleFileTar(path, txt string) (string, error) {
-	tarFile, err := ioutil.TempFile("", "create-single-file-tar-path")
+type LayerOption string
+
+const BaseLayerOption LayerOption = "baselayer"
+const WindowsOSOption LayerOption = "windows"
+
+func CreateSingleFileLayerTar(path, txt string, opts ...LayerOption) (string, error) {
```

Comment:
The name `CreateSingleFileLayerTar` is a bit misleading when `BaseLayerOption` is passed. Instead of reusing this layer generating helper, maybe we could create a helper for generating base images or something like that.

Reply:
Alternatively if we can create image (but not a container) w/o a base layer, we could change our tests to stop creating containers just to copy files. It would probably be better to assert on the `inspect.RootFs` to make sure a layer is added, instead of copying a file from the layer out of a container.

Reply:
I like the idea of a helper to generate minimal `FROM scratch`-type base images to satisfy whatever daemon. Then we still keep the option of pull/run/push using the `docker.Client`. We had a few other things sprinkled around that could be dried up by that (a default `CMD` comes to mind). I'll do a pass at that here in a new commit.

Reply:
I ended up struggling a little bit with a good helper as I really liked the comprehensibility of the basic `.NewImage(<basic deps>)` calls in the tests and any helper that spits out valid base images would mean we'd have to change them to `.NewImage(<basic deps>, .FromBaseImage(validBase))`. I also tried a helper that takes a new-potentially-invalid image and makes it valid by adding base layers but those felt too magic too. I'll tag the more straightforward, yet more verbose approach I landed on.